examples/albef/finetune_retrieval.py

```diff
@@ -0,0 +1,304 @@
+            text_to_image_scores, op=torch.distributed.ReduceOp.SUM
+        )
+
+    return image_to_text_scores.cpu(), text_to_image_scores.cpu()
+
+
+@torch.no_grad()
+def itm_eval(
+    image_to_text_scores,
+    text_to_image_scores,
+    image_to_text_mapping,
+    text_to_image_mapping,
+):
+    # Images to Text
+    ranks = torch.zeros(image_to_text_scores.size(0))
+    for index, score in enumerate(image_to_text_scores):
+        inds = torch.flip(torch.argsort(score), dims=[0])
+        rank = 1e10
+        # each image has multiple text mappings
+        # check retrieved inds with each ground truth mappping i
+        for i in image_to_text_mapping[index]:
+            tmp = torch.where(inds == i)[0][0]
+            if tmp < rank:
+                rank = tmp
+        ranks[index] = rank
+
+    # Compute metrics
+    tr1 = 100.0 * len(torch.where(ranks < 1)[0]) / len(ranks)
+    tr5 = 100.0 * len(torch.where(ranks < 5)[0]) / len(ranks)
+    tr10 = 100.0 * len(torch.where(ranks < 10)[0]) / len(ranks)
+
+    # Text to Images
+    ranks = torch.zeros(text_to_image_scores.size(0))
+    for index, score in enumerate(text_to_image_scores):
+        inds = torch.flip(torch.argsort(score), dims=[0])
+        ranks[index] = torch.where(inds == text_to_image_mapping[index])[0][0]
+
+    # Compute metrics
+    ir1 = 100.0 * len(torch.where(ranks < 1)[0]) / len(ranks)
+    ir5 = 100.0 * len(torch.where(ranks < 5)[0]) / len(ranks)
+    ir10 = 100.0 * len(torch.where(ranks < 10)[0]) / len(ranks)
+
+    tr_mean = (tr1 + tr5 + tr10) / 3
+    ir_mean = (ir1 + ir5 + ir10) / 3
+    r_mean = (tr_mean + ir_mean) / 2
+
+    eval_result = {
+        "txt_r1": tr1,
+        "txt_r5": tr5,
+        "txt_r10": tr10,
+        "txt_r_mean": tr_mean,
+        "img_r1": ir1,
+        "img_r5": ir5,
+        "img_r10": ir10,
+        "img_r_mean": ir_mean,
+        "r_mean": r_mean,
+    }
+    return eval_result
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--config", default="./examples/albef/configs/retrieval.yaml")
+    parser.add_argument("--device", default="cuda")
+    parser.add_argument(
+        "--world_size", default=1, type=int, help="number of distributed processes"
+    )
+    parser.add_argument(
+        "--dist_url", default="env://", help="url used to set up distributed training"
+    )
+    args = parser.parse_args()
+    config = yaml.load(open(args.config, "r"), Loader=yaml.Loader)
+
+    dist_utils.init_distributed_mode(args)
+    device = torch.device(args.device)
+
+    seed = 42 + dist_utils.get_rank()
+    torch.manual_seed(seed)
+    random.seed(seed)
+    cudnn.benchmark = True
+
+    datamodule = RetrievalDataModule(**config["datamodule_args"])
+    model = albef_model_for_retrieval(config)
+
+    pretrained_checkpoint_path = os.path.join(
+        config["checkpoint_root"], config["pretrained_checkpoint"]
+    )
+    model.load_state_dict(torch.load(pretrained_checkpoint_path, map_location="cpu"))
```

Comment:
In the future you will have to move to torch.hub.load_state_dict_from_url once the checkpoint is on AWS